### PR TITLE
[sailfishos][embedlite] Provide KeyNameIndex and CodeNameIndex for WidgetKeyboardEvent. Contributes to JB#55108

### PR DESCRIPTION
**Pull request by paveltkv**

### CODE REVIEW

**Comment by paveltkv** [comment]
The full set of PRs related to this:
1. https://github.com/sailfishos/qtmozembed/pull/4
2. https://github.com/sailfishos/gecko-dev/pull/36 (this)
3. https://github.com/sailfishos/sailfish-browser/pull/904

File: embedding/embedlite/embedshared/EmbedLiteViewBaseChild.cpp [hunk 1]
```diff
@@ -1110,17 +1110,58 @@ mozilla::ipc::IPCResult EmbedLiteViewBaseChild::RecvHandleTextEvent(const nsStri
   return IPC_OK();
 }
 
+static KeyNameIndex getKeyNameIndexByDomKeyCode(int domKeyCode)
+{
+#define KEY(key_, _codeNameIdx, _keyCode, _modifier)
+#define CONTROL(keyNameIdx_, _codeNameIdx, _keyCode) \
+  if (domKeyCode == _keyCode) return KEY_NAME_INDEX_##keyNameIdx_;
+#include "KeyCodeConsensus_En_US.h"
```
Comment (okodron) [inline embedding/embedlite/embedshared/EmbedLiteViewBaseChild.cpp:1118]: @rainemak, what do you think - is it ok to use this file in such way?

**Review by okodron — APPROVED** [review]

File: embedding/embedlite/embedshared/EmbedLiteViewBaseChild.cpp [hunk 2]
```diff
@@ -1110,17 +1110,58 @@ mozilla::ipc::IPCResult EmbedLiteViewBaseChild::RecvHandleTextEvent(const nsStri
...
-  event.mModifiers = Modifiers(gmodifiers);
+  event.mModifiers = Modifiers(gmodifiers) | getModifiersByCharCode(charCode);
   event.mKeyCode = charCode ? 0 : domKeyCode;
   event.mCharCode = charCode;
   event.mLocation = eKeyLocationStandard;
   event.mRefPoint = LayoutDeviceIntPoint(0, 0);
   event.mTime = PR_IntervalNow();
+  event.mKeyNameIndex = getKeyNameIndexByDomKeyCode(domKeyCode);
+  event.mKeyValue.Assign(charCode);
+  event.mCodeNameIndex = getCodeNameIndexByCharCode(charCode);
```
Comment (rainemak) [inline embedding/embedlite/embedshared/EmbedLiteViewBaseChild.cpp:1163]: Have you checked https://github.com/sailfishos/gecko-dev/blob/master/widget/NativeKeyToDOMCodeName.h and https://github.com/sailfishos/gecko-dev/blob/master/widget/NativeKeyToDOMKeyName.h ?

I'd check KeymapWrapper::InitKeyEvent from nsGtkKeyUtils.cpp

Comment (paveltkv) [reply embedding/embedlite/embedshared/EmbedLiteViewBaseChild.cpp:1163]: CODE_MAP_X11 from NativeKeyToDOMCodeName.h cannot be used as it requires scancode, but only charCode is provided for character input (in QMozViewPrivate::inputMethodEvent). Also, not critical, but NativeKeyToDOMCodeName.h does not contain MODIFIER_SHIFT matches.
NativeKeyToDOMKeyName.h contains mapping between Qt::Key and widget::KeyNameIndex. Qt::Key in our case is available only on the qtmozembed side. So I also don't see a convenient way to use it.

Comment (rainemak) [reply embedding/embedlite/embedshared/EmbedLiteViewBaseChild.cpp:1163]: We could use Qt in embedding/embedlite if that help even though target is to get rid using it.

**Review by rainemak — APPROVED** [review]

**Pull request merged**